cmd/agent/api/internal/agent/agent.go

```diff
@@ -266,7 +267,7 @@ func streamLogs(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func getDogstatsdStats(w http.ResponseWriter, r *http.Request, dogstatsdServer dogstatsdServer.Component) {
+func getDogstatsdStats(w http.ResponseWriter, r *http.Request, dogstatsdServer dogstatsdServer.Component, serverDebug dogstatsdDebug.Component) {
```

Comment:
I wonder if it would make sense to get the debug component instance from the dogstatsdServer instance, rather than supplying it independently. This would make sure we're using the right instance, rather than was provided, but isn't really connected to the actual dogstatsd server.

Reply:
I can see it both ways. I am in the camp of removing unnecessary API surface from the component, and components should be accessed via constructor injection, which means we have to pass the instances down the chain like this. 

But adding an accessor on the sever component would make this bit of code slightly simpler. Perhaps @ogaca-dd has an opinion?